### ISSUE TITLE
Reloading configuration file

### ISSUE THREAD

**Issue by daurer** [SPIhub/hummingbird#3]
It should be possible to reload the backend configuration file on the fly, without the need to restart.

----------------------------------------
- Bitbucket: https://bitbucket.org/spinitiative/hummingbird.git/issue/3
- Originally reported by: Filipe Maia
- Originally created at: 2015-02-20T11:23:26.176

**Comment by daurer** [comment]
This could be dangerous. We have tried this in the past where parameters could be adjusted on the gui then sent via ZMQ to the master. In this case one needs to ensure that all accumulated variables are flushed and that the gui knows when it is receiving data derived from the new config parameters. 

So in the end we needed to do a "soft reset" of the backend and the gui. However a "hard" reset, via our old friend Control-C, requires no change to our code base at all. I wonder if an automated reset would be a good compromise?

----------------------------------------
Original comment by: [andyofmelbourne](http://bitbucket.org/andyofmelbourne)
